app/models/v2/system.rb

```diff
@@ -65,6 +65,13 @@ class System < ApplicationRecord
         conditions: os_minor_versions(val.split.map(&:to_i), %w[IN =].include?(op)).arel.where_sql.sub(/^where /i, '')
       }
     end
+    searchable_by :assigned_or_scanned, %i[eq] do |_key, _op, _val|
+      ids = V2::System.where(id: V2::PolicySystem.select(:system_id)).or(
+        V2::System.where(id: V2::TestResult.select(:system_id))
+      ).except(:select).select(:id)
```

Comment:
Is the `except` required? Isn't it overwritten?

Reply:
Never mind. `select` appends fields. Anyhow, there is a `reselect` method that would do both (unscope and select).